lib/stringify.js

```diff
@@ -43,9 +49,9 @@ var stringify = function stringify(object, prefix, generateArrayPrefix, strictNu
 
     if (typeof obj === 'string' || typeof obj === 'number' || typeof obj === 'boolean' || utils.isBuffer(obj)) {
         if (encoder) {
-            return [encoder(prefix) + '=' + encoder(obj)];
+            return [encoder(prefix) + '=' + formatter(encoder(obj))];
```

Comment:
does the formatter only need to apply to values, or should it also apply to keys?